pkg/reconciler/service/action.go

```diff
@@ -10,11 +13,15 @@ import (
 )
 
 type ActionContext struct {
-	KubeClient       kubernetes.Client
-	WorkspaceFactory *workspace.Factory
-	Context          context.Context
-	Logger           *zap.SugaredLogger
-	ChartProvider    *chart.Provider
+	KubeClient         kubernetes.Client
+	WorkspaceFactory   *workspace.Factory
+	Context            context.Context
+	Logger             *zap.SugaredLogger
+	ChartProvider      *chart.Provider
+	Kubeconfig         string
+	InClusterClientSet k8s.Interface
```

Comment:
Isn't this part specific to the connectivity proxy? Not sure if we really need it also in the `ActionContext`to make it accessible for all other component reconcilers.

Reply:
To be handled on separate branch.

Reply:
Refactored.